OpenRA.Game/Map/CellLayer.cs

```diff
@@ -45,9 +45,23 @@ public void CopyValuesFrom(CellLayer<T> anotherLayer)
 			Array.Copy(anotherLayer.entries, entries, entries.Length);
 		}
 
+		public static CellLayer<T> CreatePrototype(Func<int, int, T> creator, int mapWidth, int mapHeight, TileShape tileShape)
```

Comment:
Forgot to escape my angle brackets :) (fixed now).

I'm suggesting ```Func<MPos, T>``` so it's clear what co-ordinate system you expect in the method.